app/scripts/modules/core/src/application/service/applicationDataSource.ts

```diff
@@ -389,6 +444,12 @@ export class ApplicationDataSource implements IDataSourceConfig {
     this.active = false;
   }
 
+  private updateData(data: any) {
```

Comment:
Is data ever not a `{}`?

Reply:
I think data is usually a `any[]`.
I intend to future PR a generic data type `T`, i.e., `ApplicationDataSource<T>`